dask/dataframe/core.py

```diff
@@ -3609,6 +3609,12 @@ class DataFrame(_Frame):
     _token_prefix = "dataframe-"
     _accessors = set()
 
+    def __init__(self, dsk, name, meta, divisions):
+        super().__init__(dsk, name, meta, divisions)
+        self.dask.layers[name].type = type(self)
+        self.dask.layers[name].divisions = divisions
+        self.dask.layers[name].chunk_type = type(meta)
```

Comment:
There's no `dtype` attribute for dataframes.

Do we want to include information about the type of what's contained in the Dask array? (If the answer here is yes, then I can add something similar for arrays in this PR)